src/sec.c

```diff
@@ -23,10 +23,15 @@ sec_bin_init(sec_bin_t *bin) {
 bool
 sec_init(tsdn_t *tsdn, sec_t *sec, base_t *base, pai_t *fallback,
     const sec_opts_t *opts) {
-	size_t max_alloc = opts->max_alloc & PAGE_MASK;
-	pszind_t npsizes = sz_psz2ind(max_alloc);
-	if (sz_pind2sz(npsizes) > opts->max_alloc) {
-		npsizes--;
+	assert(opts->max_alloc > 0);
+
+	pszind_t npsizes = sz_psz2ind(opts->max_alloc);
```

Comment:
`sz_psz2ind` kinda reads like `page size to ind` to me, so maybe let's only pass page sizes to it? Plus this is an init function so the extra `& ~PAGE_MASK` doesn't really hurt.

Reply:
I agree, will change.